corehq/apps/domain/decorators.py

```diff
@@ -241,6 +241,22 @@ def __inner__(request, domain, *args, **new_kwargs):
     return decorator
 
 
+def api_domain_view(view):
+    """
+    Decorate this with any domain view that should be accessed via api only
+    """
+    @wraps(view)
+    @api_key()
+    @login_and_domain_required
```

Comment:
The names are little confusing, `login_and_domain_required` checks only if `request.user.is_authenticated()` is true or not, it doesn't check how the authentication occurred. The `api_key` decorator that's above this decorator ensures only api-key auth goes through.

I have also tested it against CSRF/session auth, as expected it results in `401`